mifosng-android/src/main/java/com/mifos/mifosxdroid/online/generatecollectionsheet/GenerateCollectionSheetPresenter.java

```diff
@@ -83,44 +118,65 @@ public void onCompleted() {
 
                     @Override
                     public void onError(Throwable e) {
-                        getMvpView().showFetchingError("Failed to fetch Staff in Office");
+                        getMvpView().showProgressbar(false);
+                        try {
+                            if (e instanceof HttpException) {
+                                String errorMessage = ((HttpException) e).response().errorBody()
+                                        .string();
+                                getMvpView().showError(MFErrorParser.parseError(errorMessage)
+                                        .getErrors().get(0).getDefaultUserMessage());
+                            }
+                        } catch (Throwable throwable) {
+                            RxJavaPlugins.getInstance().getErrorHandler().handleError(e);
+                        }
                     }
 
                     @Override
                     public void onNext(List<Staff> staffs) {
+                        getMvpView().showProgressbar(false);
                         getMvpView().showStaffInOffice(staffs, officeId);
                     }
-                });
+                }));
     }
 
     public void loadCentersInOffice(int id, Map<String, Object> params) {
         checkViewAttached();
-        if (mSubscription != null) mSubscription.unsubscribe();
-        mSubscription = mDataManager.getCentersInOffice(id, params)
+        getMvpView().showProgressbar(true);
+        mSubscription.add(mDataManager.getCentersInOffice(id, params)
                 .observeOn(AndroidSchedulers.mainThread())
                 .subscribeOn(Schedulers.io())
                 .subscribe(new Subscriber<List<Center>>() {
                     @Override
                     public void onCompleted() {
-
                     }
 
                     @Override
                     public void onError(Throwable e) {
-                        getMvpView().showFetchingError("Failed to fetch Centers in Office");
+                        getMvpView().showProgressbar(false);
+                        try {
```

Comment:
use this utils method to show error https://github.com/openMF/android-client/blob/master/mifosng-android/src/main/java/com/mifos/utils/MFErrorParser.java#L25